src/internal/item-container/index.tsx

```diff
@@ -399,13 +440,52 @@ function ItemContainerComponent(
     }
   }
 
-  const dragHandleRef = useRef<HTMLButtonElement>(null);
+  const dragHandleRef = useRef<HTMLDivElement>(null);
   useImperativeHandle(ref, () => ({
-    focusDragHandle: () => dragHandleRef.current?.focus(),
+    focusDragHandle: () => {
+      return dragHandleRef.current?.focus();
+    },
   }));
 
+  const hookProps: UseInternalDragHandleInteractionStateProps<HandleOperation> = {
+    onDndStartAction: (event, handleOperation) => {
+      if (handleOperation === "drag") {
+        onDragHandleDndTransitionStart(event);
+        return;
+      }
+      onResizeHandlePointerDown(event);
```

Comment:
Why do we use different logic here for drag and resize?

Reply:
Because the calculation inside the function depends on the operation:
- Resize: we calculate the offset between item's **bottom-right corner** and the pointer landing position.
- Drag: we calculate the offset between item's **top-left corner** and the pointer landing position.

Thare are quiet some things common in both functions. I've extracted the logic in a pure function and put it in a separate file + tests.